clients/include/host_batch_vector.hpp

```diff
@@ -148,7 +188,82 @@ class host_batch_vector
         }
     };
 
+    //!
+    //! @brief Initialize with the rocblas random number generator.
+    //! @param seedReset if true reset the seed.
+    //!
+    inline void random_init(bool seedReset = true) noexcept
```

Comment:
Not sure if the memory helpers should introduce dependency on random.   Other random helpers exist so could handle the batch_vectors there?

Reply:
Yeah, I'm not sure why random number stuff is here.

Also, `noexcept` is mostly redundant nowadays.

Reply:
This is utility classes for tests. Why using random is an issue?..

Reply:
Not an issue, but putting all the factory like initializers in a separate file might be a cleaner pattern to follow.   Keeping the memory by itself seems like it is a lot of code already, especially if we add in irregular batch helpers, maybe they will require the random, but could be a separate class/file anyway.